### PR TITLE
fix(android): Be able to disable the subtitles

### LINKED ISSUE #128: Unable to turn off subtitle
**Description**
Users are currently only able to turn off subtitles in player if subtitle is set to be off in profile page

**Expected behaviour**
Turning off subtitle in player should turn off subtitle regardless default setting in profile page

This only occurs for Android

### PR DESCRIPTION
**Pull request by cj-wessels**
## Discussion points

Fixed #128

### CODE REVIEW

File: external_projects/bccm_player/android/src/main/kotlin/media/bcc/bccm_player/players/exoplayer/ExoPlayerController.kt [hunk 1]
```diff
@@ -111,9 +113,9 @@ class ExoPlayerController(private val context: Context) :
             "bccm",
             "setting preferred audio and sub lang to: ${appConfigState?.audioLanguage}, ${appConfigState?.subtitleLanguage}"
         )
-        player.trackSelectionParameters = trackSelector.parameters.buildUpon()
-            .setPreferredAudioLanguage(appConfigState?.audioLanguage)
```
Comment (andreasgan) [inline external_projects/bccm_player/android/src/main/kotlin/media/bcc/bccm_player/players/exoplayer/ExoPlayerController.kt:115]: Looks like you removed audio selection by accident

File: external_projects/bccm_player/android/src/main/kotlin/media/bcc/bccm_player/players/exoplayer/ExoPlayerController.kt [hunk 2]
```diff
@@ -186,6 +190,35 @@ class ExoPlayerController(private val context: Context) :
...
 
+    /**
+     * ExoPlayers `setPreferredTextLanguage` makes it impossible to
+     * disable the subtitle.
+     *
+     * Instead of `setPreferredLanguage` we force the text track to default languag (from settings)
+     * once. This method handles this behavior.
+     */
+    private fun setDefaultTextTrack(tracks: Tracks) {
+        if (textLanguageThatShouldBeSelected == null) {
```
Comment (andreasgan) [inline external_projects/bccm_player/android/src/main/kotlin/media/bcc/bccm_player/players/exoplayer/ExoPlayerController.kt:201]: Suggestion
- rename this to setTextTrack
- move the "run only once"-logic into `onTracksChanged`

Comment (andreasgan) [reply external_projects/bccm_player/android/src/main/kotlin/media/bcc/bccm_player/players/exoplayer/ExoPlayerController.kt:201]: would make it more readable whats going on

**Review by andreasgan — CHANGES_REQUESTED** [review]
Great that you found a solution 🎉

**Comment by andreasgan** [comment]
Fixes #128

**Review by andreasgangso — APPROVED** [review]

**Pull request merged**